Glossary created. First batch of terms added. (#45)
sporeprotocol/spore-docs: docs/glossary/terms.md

---
sidebar_position: 1
sidebar_label: Terms
---

> 

Reviewer: Maybe add a title? something like: "# Terms"

Author: > Maybe add a title? something like: "# Terms"

added

> ## anyone-can-pay (ACP)

Reviewer: Why are the initial letters of the terms all in lowercase? Should they be capitalized instead?
For example: "anyone-can-pay (ACP)" -> "Anyone-can-pay (ACP)"

Author: Regarding capitalization there's no strict rule. The key is to keep the style consistent.

Reviewer 2: I vote for capitalizing the first letter of every term, the current mix is awkward

## content-type
Describes the file format or type of the digital asset associated with the token. For instance, content-type like JPEG, MP3, MP4 specify that the content is an image, audio, or video, respectively, among other formats. This information helps determine how to handle and render the associated content. Its scheme and rules follows the [RFC of MIME](https://datatracker.ietf.org/doc/html/rfc1341) standard.

**see also:**
* [content](#content)
* [Why combine content and content-type?](https://docs.spore.pro/basics/tech-faq#why-combine-content-and-content-type)

## destroy

Refer to the term [melt](#melt).

**see also:** 
* [mint](#mint)
* [transfer](#transfer)

## extension

Specified as Spore [Mutant](#mutant), used to add extensional behavior or verify rules onto spore or cluster.

**see also:** 
* [mutant](#mutant)

## group

Collection of individual Spores associated together by a specific cluster.

**see also:** 
* [cluster](#cluster)

## immortal

Extension making a spore or a cluster permanently indestructible on-chain, impossible to be destroyed or remove from the chain, even by their owners.

## immutability

> Quality of permanence and resistance to alteration. Once a Spore is minted, its essential attributes, including ownership, provenance, and metadata, become fixed and cannot be changed. This guarantees the security and integrity of these attributes on the blockchain.

Reviewer 2: ```suggestion
Quality of permanence and resistance to alteration. Once a Spore is minted, its essential attributes become fixed and cannot be changed. 
```

Omnilock introduces a regulation compliance module, enabling administrators to revoke tokens held by users under defined circumstances, bridging the gap between blockchain and traditional financial regulation. 

For more information, visit [RFC0042-Omnilock](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0042-omnilock/0042-omnilock.md).

## output

Spore cell(s) created when new transactions are confirmed. Can serve as input cells for future transactions.

**see also:** 
* [input](#input)

## on-chain ownership

Publicly verifiable and indisputable record of ownership for digital assets inscribed on the blockchain, eliminating the need for intermediaries and ensuring transparency.

## private cluster

Cluster permitting only the owner to associate a Spore or multiple Spores when minting.

**see also:** 
* [public cluster](#public_cluster)
* [cluster](#cluster)

## private key

Cryptographic information exclusively known to its owner or owners. Employed in conjunction with an algorithm for both encryption and decryption, ensuring the confidentiality and security of sensitive information.

**see also:** 
* [public key](#public_key)

## public cluster

> Cluster enabling anyone, besides the owner, to associate a Spore or multiple Spores when minting.

Reviewer 2: ```suggestion
A cluster that allows anyone, not just the owner, to associate a Spore or multiple Spores with it when minting Spores.
```
"besides the owner" excludes the owner

**see also:**
* [cluster](#clsuter)
* [private cluster](#private_cluster)

## public key

Cryptographic information openly shareable without compromising security. Utilized for message encryption to ensure that only the corresponding private key holder can decrypt. Also employed for message authenticity verification, confirming authorization by the user possessing the corresponding private key.

**see also:** 
* [private key](#public_key)

## redeemability

Ability of Spore users to destroy their Spores and instantly recover the remaining value at no cost. 

**see also:**
* [Why is Spore‘s value redeemable?](https://docs.spore.pro/basics/tech-faq#why-is-spores-value-redeemable)

## secp256k1-blake160

A lock script code using the [secp256k1](https://en.bitcoin.it/wiki/Secp256k1) signature algorithm, performing [Blake2b](https://www.blake2.net/) hash operations on transaction and witness components. Essential for validating transactions in specific blockchain systems.

## transfer

> Process that refreshes a Spore by consuming its cell and creating a new one with the same Spore ID and unchanged fields. This operation costs no additional free, where the costs is deducted from the cell's capacity.

Reviewer 2: ```suggestion
The process of changing the ownership of a Spore or a Cluster. It's important to note that owning a cluster does not automatically grant ownership of the spores associated with that cluster. Each Spore and Cluster has its own distinct ownership rights that can be transferred separately.
```
the implementation bits should be sufficient under "see also", especially the information is the same as basics/spore-operation#transfer
